routes/login/login.controller.js

```diff
@@ -39,19 +39,11 @@ const postLoginCode = (req, res) => {
 }
 
 const postSIN = (req, res) => {
-
-  //If sin is not set, set it to null
-  let sin = req.body.sin || null
-  req.session.sin = sin
-
   const errors = validationResult(req)
 
   if (!errors.isEmpty()) {
-    // clear session
-    req.session = null
-
     return res.status(422).render('login/sin', {
-      data: { sin: req.body.sin } || {},
+      data: { ...req.session, ...{ sin: req.body.sin } } || {},
```

Comment:
What are the benefits of passing it as a spread instead of as an array?

Reply:
Spreads let you merge dictionaries — arrays don't let you do that.

So, for example:

```js
var dict1 = {
  a: 'A',
  b: 'B'
}

var dict2 = { ...dict1, ...{ c: 'C' } }

/* 

 dict2 looks like this 

{
  a: 'A',
  b: 'B',
  c: 'C'
}
*/
```

In this case, we need to pass in 

- the value that the user gave us (so we can put it back in the input)
- the good value (ie, for the hint text)

So we merge both dictionaries together.